libraries/abstractions/common_io/test/test_iot_uart.c

```diff
@@ -399,44 +402,41 @@ TEST( TEST_IOT_UART, AFQP_IotUARTWriteAsyncReadAsyncLoopbackTest )
 TEST( TEST_IOT_UART, AFQP_AssistedIotUARTWriteAsync )
 {
     IotUARTHandle_t xUartHandle;
-    int32_t lIoctl, lWrite, lClose, lRead, lTransferAmount_1, lTransferAmount_2;
+    int32_t lIoctl, lWrite, lClose, lRead, lTransferAmount;
     uint8_t i;
     uint8_t cpBufferLarge[ testIotUART_BUFFER_LENGTH_LARGE + 1 ] = { 0 };
-    uint8_t cpBufferRead[ testIotUART_BUFFER_LENGTH_LARGE + 1 ] = { 0 };
-    uint8_t uStringCompare = 1;
+    BaseType_t xCallbackReturn;
 
     for (i = 0; i < testIotUART_BUFFER_LENGTH_LARGE; i++)
     {
         cpBufferLarge[i] = 0xAA;
     }
+    strncpy(&cpBufferLarge[testIotUART_BUFFER_LENGTH_LARGE ], "\n", 1);
 
     xUartHandle = iot_uart_open( ustestIotUartPort );
     TEST_ASSERT_NOT_EQUAL( NULL, xUartHandle );
 
+    iot_uart_set_callback( xUartHandle, prvReadWriteCallback, NULL );
+
+
     if( TEST_PROTECT() )
     {
         lWrite = iot_uart_write_async( xUartHandle, cpBufferLarge, testIotUART_BUFFER_LENGTH_LARGE );
         TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lWrite );
 
-        lIoctl = iot_uart_ioctl( xUartHandle, eGetTxNoOfbytes, &lTransferAmount_1 );
-        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
-        TEST_ASSERT_GREATER_THAN( 0, lTransferAmount_1  );
-
-        /* Delay for 1 sec. */
-        vTaskDelay(testIotUART_DELAY);
+        xCallbackReturn = xSemaphoreTake( xtestIotUARTSemaphore, testIotUART_DEFAULT_SEMPAHORE_DELAY );
+        TEST_ASSERT_EQUAL( pdTRUE, xCallbackReturn );
 
-        lIoctl = iot_uart_ioctl( xUartHandle, eGetTxNoOfbytes, &lTransferAmount_2 );
+        lIoctl = iot_uart_ioctl( xUartHandle, eGetTxNoOfbytes, &lTransferAmount );
```

Comment:
Same question here. Why is this moved after read is completed? At this point lTranferAmount should be the size of the buffer anyway. 
Also earlier, we were making sure this can be checked multiple times and it increases

Reply:
Will revert this too